ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt

```diff
@@ -145,13 +145,27 @@ constructor(
   }
 
   suspend fun enableLocationLockAndGetUpdates() {
-    permissionsManager.obtainPermission(Manifest.permission.ACCESS_FINE_LOCATION)
-    settingsManager.enableLocationSettings(FINE_LOCATION_UPDATES_REQUEST)
-    enableLocationLock()
-    locationManager.requestLocationUpdates()
+    try {
+      permissionsManager.obtainPermission(Manifest.permission.ACCESS_FINE_LOCATION)
+      settingsManager.enableLocationSettings(FINE_LOCATION_UPDATES_REQUEST)
+      enableLocationLock()
+      locationManager.requestLocationUpdates()
+    } catch (throwable: ApiException) {
+      val statusCode = throwable.statusCode
+      if (statusCode == SETTINGS_CHANGE_UNAVAILABLE) {
+        Timber.e(
+          throwable,
+          "User is offline, so fallback to user's current permission, which may also fail."
+        )
+        enableLocationLock()
+        locationManager.requestLocationUpdates()
```

Comment:
Can we just move these two lines after the catch block and remove the duplicate lines from the try block?

Reply:
Good call! Done, and coordinated with the change that @shobhitagarwal1612 suggested below.